src/saveImpl.ts

```diff
@@ -47,13 +48,18 @@ export async function saveImpl(
         // NO-OP in case of SaveOnly action
         const restoredKey = stateProvider.getCacheState();
 
-        if (utils.isExactKeyMatch(primaryKey, restoredKey)) {
+        const forceOverwrite = utils.getInputAsBool(Inputs.ForceOverwrite);
+        if (utils.isExactKeyMatch(primaryKey, restoredKey) && !forceOverwrite) {
             core.info(
-                `Cache hit occurred on the primary key ${primaryKey}, not saving cache.`
+                `Cache hit occurred on the primary key ${primaryKey} and force-overwrite is disabled, not saving cache.`
             );
             return;
         }
 
+        if (utils.isExactKeyMatch(primaryKey, restoredKey) && forceOverwrite) {
+            await issueCommand('actions-cache delete', {}, primaryKey);
```

Comment:
@PrinsFrank were you able to validate these changes in your own repository?

I'm not entirely sure if the backend supporting the cache action will allow overwriting an existing cache entry like this, and as far as I know this `actions-cache delete` command is not a real command.

Reply:
Sorry, I got confused with the CI options that are available in the actions-cache plugin, but not in the toolkit yet. I use that here: https://github.com/PrinsFrank/CI-PHP/blob/main/.github/workflows/quality.yml#L134, so I can confirm this workflow works, the code just currently doesn't. I wasn't sure how to run this code, but I figured it out and can see this doesn't work in the current form.

If I understand correctly there's currently not any code for this at all in upstream packages, so I tried implementing it in https://github.com/actions/toolkit/pull/1783. As I mentioned before, I'm not a typescript developer so any pointers would be very much appreciated.